kernel/safe_typing.ml

```diff
@@ -771,35 +742,41 @@ let export_side_effects mb env c =
         | [] -> List.rev acc, ce
         | eff :: rest ->
           if Int.equal sl 0 then
-           let env, cb =
-              let { seff_constant = kn; seff_body = ocb; seff_env = u ; _ } = eff in
-               let ce = constant_entry_of_side_effect ocb u in
+            let env, cb =
+              let kn = eff.seff_constant in
+               let ce = constant_entry_of_side_effect eff in
                let cb = Term_typing.translate_constant Term_typing.Pure env kn ce in
-               let eff = { eff with
-                seff_body = cb;
-                seff_env = `Nothing;
-               } in
+               let cb = map_constant (fun c -> Future.force c) cb in
```

Comment:
Why eta expand?

Reply:
A reflex of mine in OCaml to be sure to control the arity of the function. Doesn't hurt either, and is future-proof if we start messing with the internal type of opaque definitions.

Reply:
>A reflex of mine in OCaml to be sure to control the arity of the function

How does eta expanding do that?

Reply:
The discipline is to always write fully applied functions, but that's not enforced by the compiler, it's merely a reminder to the reader. I personally cannot stand the point-free style that is commonplace in Haskell for instance, call me a C programmer if you want.

Reply:
It's not partially applied here though

Reply:
The analogy with `C` also makes little sense to me. It is quite the contrary in fact, you'd pass a pointer to the function here, not an anonymous function... Please fix.

Reply:
*Seriously* guys? Sigh.

Reply:
If a painter is allowed to have a blue period, maybe a coder should be allowed to have an eta-long period. ;-) Or maybe not...